test/test_tree_1d.jl

```diff
@@ -474,6 +474,170 @@ end # SWE
+                                0.007355554947688914,
+                                0.0021140505291696865,
+                                0.001562655691777548,
+                                0.002409077213895827,
+                                0.0002187445586192549,
+                            ],
+                            tspan=(0.0, 0.25))
+        # Ensure that we do not have excessive memory allocations
+        # (e.g., from type instabilities)
+        let
+            t = sol.t[end]
+            u_ode = sol.u[end]
+            du_ode = similar(u_ode)
+            @test (@allocated Trixi.rhs!(du_ode, u_ode, semi, t)) < 1000
+        end
+    end
+
+    @trixi_testset "elixir_shallowwater_multilayer_convergence.jl with flux_lax_friedrichs" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "elixir_shallowwater_multilayer_convergence.jl"),
+                            l2=[
+                                0.0019005751369889307,
+                                0.0006416934635148739,
+                                0.001323748067661235,
+                                0.0002514848406032196,
+                                0.00019551914601560465,
+                                0.00029396199135642574,
+                                0.00013912637728693124,
+                            ],
+                            linf=[
+                                0.005947927726240643,
+                                0.0028032845734302647,
+                                0.003944642659854947,
+                                0.0009866355929912807,
+                                0.0009301080749861135,
+                                0.0012826136652659414,
+                                0.0002187445586192549,
+                            ],
+                            surface_flux=(flux_lax_friedrichs,
+                                          flux_nonconservative_ersing_etal),
+                            tspan=(0.0, 0.25))
+        # Ensure that we do not have excessive memory allocations
+        # (e.g., from type instabilities)
+        let
+            t = sol.t[end]
+            u_ode = sol.u[end]
+            du_ode = similar(u_ode)
+            @test (@allocated Trixi.rhs!(du_ode, u_ode, semi, t)) < 1000
+        end
+    end
+
+    @trixi_testset "elixir_shallowwater_multilayer_well_balanced.jl" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "elixir_shallowwater_multilayer_well_balanced.jl"),
+                            l2=[
+                                1.2015338985485869e-17,
+                                1.4450281975802518e-17,
+                                0.001002881985401688,
+                                5.6968770683622844e-18,
+                                5.8630384557660415e-18,
+                                1.938143907661441e-17,
+                                0.0010028819854016856,
```

Comment:
Interesting, it might be worthwhile to set the bottom topography to be a polynomial instead of a cosine function. If the projection error is the culprit then these two errors should become much better.

Reply:
To test this I ran a well-balanced test with a quadratic polynomial as bottom topography and now the residuals are exact.

![image](https://github.com/trixi-framework/TrixiShallowWater.jl/assets/114223904/a74f0fac-250f-4583-9a4c-bba685f75e8d)

Overall, this shouldn't be a concern though, as we get well-balancedness for both cases.

Reply:
Okay, as expected. There is no concern, I was just curious that I was interpreting the errors correctly.